tests/keras/test_callbacks.py

```diff
@@ -936,5 +938,44 @@ def tests_RemoteMonitorWithJsonPayload():
                   validation_data=(X_test, y_test), callbacks=cbks, epochs=1)
 
 
+def test_TensorBoard_trainable_weights(tmpdir):
+    np.random.seed(np.random.randint(1, 1e7))
+    filepath = str(tmpdir / 'logs')
+
+    input_shape = (72, 72, 3)
+    (x_train, y_train), (x_test, y_test) = get_test_data(num_train=train_samples,
+                                                         num_test=test_samples,
+                                                         input_shape=input_shape,
+                                                         classification=True,
+                                                         num_classes=num_classes)
+    y_train = np_utils.to_categorical(y_train)
+    y_test = np_utils.to_categorical(y_test)
```

Comment:
In general this test is more like an integration test than a unit test. You don't need half of this stuff:

- the model should be minimal (this one has a bunch of extra layers)
- you don't need data with a statistical structures, `np.random` will work just fine
- etc.